File: packages/dart/smoke.dart
// Live smoke check against the real snapshot API. Excluded from the published
// package, it exists so CI exercises the real HttpClient + TLS path that the
// offline test suite deliberately never touches.
import 'dart:io';

import 'package:litportnet_free_proxy/litportnet_free_proxy.dart';

Future<void> main() async {
  final client = LitportFreeProxyClient(timeout: const Duration(seconds: 20));
  try {
    final proxies = await client.pickBest(
      3,
      const ProxyFilters(protocol: 'socks5', checkedWithinMin: 1440),
    );
    stdout.writeln('live smoke: ${proxies.length} proxies');
    for (final proxy in proxies) {
      stdout.writeln('  ${proxy.url} uptime7d=${proxy.uptime7d} '
          'latencyMs=${proxy.latencyMs}');
    }
    if (proxies.isEmpty) exitCode = 1;
  } catch (error) {
    stdout.writeln('live smoke failed: $error');
    exitCode = 1;
  } finally {
    client.close();
  }
}

File: packages/r/smoke.R
# Live smoke check against the real snapshot API. Excluded from the built
# package (see .Rbuildignore); it exists so CI exercises the real curl path that
# the offline test suite deliberately never touches.
library(litportFreeProxy)

client <- litport_client(timeout = 20)
proxies <- pick_best(client, 3, protocol = "socks5", checked_within_min = 1440)

cat(sprintf("live smoke: %d proxies\n", nrow(proxies)))
if (nrow(proxies) == 0) {
  stop("live smoke returned no proxies")
}
print(proxies[, c("url", "uptime_7d", "latency_ms", "country")])
